x-pack/plugins/enterprise_search/server/lib/indices/create_index.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MappingKeywordProperty, MappingTextProperty } from '@elastic/elasticsearch/lib/api/types';
+import { IScopedClusterClient } from '@kbn/core/server';
+
+import { textAnalysisSettings } from './text_analysis';
+
+const prefixMapping: MappingTextProperty = {
```

Comment:
This is one of a couple of concessions I made to get the type checker to cooperate. When these were untyped I was getting errors.